ambry-frontend/src/main/java/com/github/ambry/frontend/FrontendRestRequestService.java

```diff
@@ -209,7 +211,7 @@ public void start() throws InstantiationException {
     if (frontendConfig.enableNamedBlobCleanupTask) {
       namedBlobsCleanupScheduler = Utils.newScheduler(1, "named-blobs-cleanup-", false);
       namedBlobsCleanupTask =
-          namedBlobsCleanupScheduler.scheduleAtFixedRate(namedBlobsCleanupRunner, 60 * 10, 60 * 10, TimeUnit.SECONDS);
+          namedBlobsCleanupScheduler.schedule(namedBlobsCleanupRunner, random.nextInt(60*60*24*7), TimeUnit.SECONDS);
```

Comment:
we still want to run this periodically, we want to run this with a delay and then periodically run it.

Reply:
Ok, got it, since I thought we had weekly release, we don't need to schedule it to run weekly. But I agree it's better to repeat weekly in case the ambry release delayed or canceled..